File: java/prime_number.java
import javax.swing.*;

class Prime {
    public static void main(String args[]) {
        int i,num,flag=1;
	num = Integer.parseInt(JOptionPane.showInputDialog("enter number "));

	for(i=2;i<num;i++)
	{
		if(num%i==0)
		{
			flag=0;
			break;
		}
	}
		if(flag==1)
		{
			System.out.println("number is prime");
		}
		else
		{
			System.out.println("number is not prime");
		}		
	 
     }
}

File: c++/prime_number.cpp
#include<iostream>
using namespace std;

int main()
{
    int flag = 1;
    int num;
    cout << "Enter a number to check if it is prime or not: ";
    cin >> num;

    if (num <= 1) {
        flag = 0; 
    } else {
        for (int i = 2; i <= num / 2; i++) 
        {
            if (num % i == 0)
            {
                flag = 0; 
                break;
            }
        }
    }

    if (flag == 1)
    {
        cout << "Number is prime.";
    }
    else
    {
        cout << "Number is not prime.";
    }

    return 0;
}
